examples/bsc-sdk/src/node/evm/executor.rs

```diff
@@ -0,0 +1,207 @@
+    gas_used: u64,
+    /// Receipts of executed transactions.
+    receipts: Vec<R::Receipt>,
+    /// System txs
+    system_txs: Vec<&'a R::Transaction>,
+    /// Receipt builder.
+    receipt_builder: R,
+}
+
+impl<'a, DB, EVM, Spec, R: ReceiptBuilder> BscBlockExecutor<'a, EVM, Spec, R>
+where
+    DB: Database + 'a,
+    EVM: Evm<DB = &'a mut State<DB>, Tx: FromRecoveredTx<R::Transaction>>,
+    Spec: EthereumHardforks + BscHardforks + EthChainSpec + Hardforks,
+    R: ReceiptBuilder<Transaction: Transaction + Encodable2718, Receipt: TxReceipt<Log = Log>>,
+{
+    /// Creates a new BscBlockExecutor.
+    pub fn new(
+        inner: EthBlockExecutor<'a, EVM, &'a Arc<ChainSpec>, &'a RethReceiptBuilder>,
+        spec: Spec,
+        evm: EVM,
+        receipt_builder: R,
+    ) -> Self {
+        Self {
+            inner,
+            spec,
+            evm,
+            gas_used: 0,
+            receipts: vec![],
+            system_txs: vec![],
+            receipt_builder,
+        }
+    }
+
+    /// Applies system contract upgrades if the Feynman fork is not yet active.
+    fn apply_upgrade_contracts_if_before_feynman(&mut self) -> Result<(), BlockExecutionError> {
+        if self.spec.is_feynman_active_at_timestamp(self.evm.block().timestamp) {
+            return Ok(());
+        }
+
+        let contracts = get_upgrade_system_contracts(
+            &self.spec,
+            self.evm.block().number,
+            self.evm.block().timestamp,
+            self.evm.block().timestamp - 3_000, // TODO: how to get parent block timestamp?
+        )
+        .map_err(|_| BlockExecutionError::msg("Failed to get upgrade system contracts"))?;
+
+        for (address, maybe_code) in contracts {
+            if let Some(code) = maybe_code {
+                self.upgrade_system_contract(address, code)?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Replaces the code of a system contract in state.
+    fn upgrade_system_contract(
+        &mut self,
+        address: Address,
+        code: Bytecode,
+    ) -> Result<(), BlockExecutionError> {
+        let account =
+            self.evm.db_mut().load_cache_account(address).map_err(BlockExecutionError::other)?;
+
+        let mut info = account.account_info().unwrap_or_default();
+        info.code_hash = code.hash_slow();
+        info.code = Some(code);
+
+        let transition = account.change(info, Default::default());
+        self.evm.db_mut().apply_transition(vec![(address, transition)]);
+        Ok(())
+    }
+}
+
+impl<'a, DB, E, Spec, R> BlockExecutor for BscBlockExecutor<'a, E, Spec, R>
+where
+    DB: Database + 'a,
+    E: Evm<DB = &'a mut State<DB>, Tx: FromRecoveredTx<R::Transaction>>,
+    Spec: EthereumHardforks + BscHardforks + EthChainSpec + Hardforks,
+    R: ReceiptBuilder<Transaction: Transaction + Encodable2718, Receipt: TxReceipt<Log = Log>>,
+    <R as ReceiptBuilder>::Transaction: Unpin,
+{
+    type Transaction = R::Transaction;
```

Comment:
this is how we define the tx type this executor operates on, which maps this to the tx type the receiptbuilder operates on.

this already enforces two trait bounds `Transaction + Encodable2718` we can easily add more, even newly defined traits, if we need access to the signature for example